### PR TITLE
Template matching key time

### PR DESCRIPTION
**Pull request by Ratoone**
Introduced generic multiscale template matching. This was applied to clef and time signature detection. Added template and sample images to the project

### CODE REVIEW

File: music_detection/staff.py [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+    def identify_clef_time(self):
+        if self.key == KeyEnum.UNDEFINED:
+            clef = pick_template(self.template_manager.clef, self.image)
+            if clef is not None:
+                self.key = clef
+
+        if self.time_signature == TimeSignatureEnum.UNDEFINED:
+            time = pick_template(self.template_manager.time_signature, self.image)
+            if time is not None:
+                self.time_signature = time
```
Comment (gormasab) [inline music_detection/staff.py:36]: pls split this in another function

File: music_detection/utils/template_matching.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    found = None
+
+    for scale in np.linspace(0.5, 1.5, 20):
+        resized_template = cv2.resize(template, None, fx=scale, fy=scale, interpolation=cv2.INTER_CUBIC)
+        if resized_template.shape[0] > image.shape[0] or resized_template.shape[1] > image.shape[1]:
+            break
+
+        result = cv2.matchTemplate(image, resized_template, cv2.TM_CCOEFF_NORMED)
+        (_, confidence, _, max_position) = cv2.minMaxLoc(result)
+        if found is None or confidence > found[0]:
```
Comment (gormasab) [inline music_detection/utils/template_matching.py]: u have to write comments in ur code everywhere, for more information look at rey's nice codes.

File: music_detection/utils/template_matching.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            found = (confidence, max_position, scale)
+
+    (score, position, scale) = found
+    return score, position
```
Comment (gormasab) [inline music_detection/utils/template_matching.py]: pls change score to some other name to avoid confusion.

**Review by gormasab — CHANGES_REQUESTED** [review]
thanks for the code. However, next time more comments are expected. You don't always accompany your code to explain it.

**Comment by jrmsgr** [comment]
Looks good to me, consider commenting your code anyway :)

**Review by jrmsgr — APPROVED** [review]

**Comment by Ratoone** [comment]
It is commented :)). Please show me where more comments are needed, and I will gladly add them

**Review by gormasab — APPROVED** [review]
I approved but I kindly request for more comments next time.

**Pull request merged**